lstchain/tools/lstchain_create_dl3_file.py

```diff
@@ -173,31 +190,74 @@ def setup(self):
 
     def start(self):
 
-        self.data = read_data_dl2_to_QTable(str(self.input_dl2))
-        self.effective_time, self.elapsed_time = get_effective_time(self.data)
-        self.run_number = run_info_from_filename(self.input_dl2)[1]
+        if not self.source_dep:
+            self.data = read_data_dl2_to_QTable(str(self.input_dl2))
+            self.effective_time, self.elapsed_time = get_effective_time(self.data)
+            self.run_number = run_info_from_filename(self.input_dl2)[1]
```

Comment:
You have the same code later in the loop. Just move outside the if here and remove from the loop code.

Reply:
thanks, done